packages/presentation/lib/screens/login/login_bloc.dart

```diff
@@ -110,18 +137,87 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
 
       _updateFieldControllers(user);
 
-      add(LoginData(
-        false,
-      ));
-
       await _loginIfUserRegistered(user);
-    } on AuthFailureException {
-      add(
-        LoginData(
-          true,
-        ),
-      );
+    } on AuthFailureException catch (e) {
+      print(e);
+    }
+  }
+
+  @override
+  GlobalKey<FormState> get formStateGlobalKey => _formStateGlobalKey;
+
+  @override
+  void validateForm() {
+    final user = UserEntity(
+      login: _loginController.text,
+      password: _passwordController.text,
+    );
+
+    final validationResult = _loginValidationUseCase(user);
+
+    add(
+      state.copyWith(
+        loginIsEmpty: validationResult.loginIsEmpty,
+        loginIsCorrect: validationResult.loginIsCorrect,
+        passwordIsEmpty: validationResult.passwordIsEmpty,
+        passwordIsCorrect: validationResult.passwordIsCorrect,
+      ),
+    );
+
+    if (_formStateGlobalKey.currentState!.validate()) {
+      onLogin();
+    }
+  }
+
+  @override
+  String? loginValidator(
+    String invalidLoginMessage,
+    String wrongLoginMessage,
+    String requiredLoginMessage,
+  ) {
+    if (state.authFailure) {
+      return invalidLoginMessage;
+    }
+
+    if (state.loginIsEmpty) {
+      return requiredLoginMessage;
+    }
+
+    if (!state.loginIsCorrect) {
+      return wrongLoginMessage;
```

Comment:
I think we can move this logic to viewMapper

Reply:
Fixed.